File: server/python/app/presentation/mappers/payment_details_mapper.py
from app.application.dtos.get_payment_details.response_dto import GetPaymentDetailsResponseDto
from app.presentation.models.get_payment_details.response import GetPaymentDetailsResponse

class PaymentDetailsPresentationMapper:

    @staticmethod
    def from_dto(dto: GetPaymentDetailsResponseDto) -> GetPaymentDetailsResponse:
        return GetPaymentDetailsResponse(
            id=dto.id,
            status=dto.status,
            status_output=dto.status_output,
            payment_output=dto.payment_output,
            hosted_checkout_specific_output=dto.hosted_checkout_specific_output,
            operations=dto.operations
        )

File: server/nodejs/src/presentation/mappers/payment-details-mapper.ts
import { GetPaymentDetailsResponseDto } from '../../business/dtos/payment/get-payment-details/get-payment-details-response-dto';
import { GetPaymentDetailsResponse } from '../models/payment/get-payment-details/get-payment-details-response';

export const PaymentDetailsMapper = {
    fromDto: (response: GetPaymentDetailsResponseDto): GetPaymentDetailsResponse => {
        return {
            id: response.id,
            status: response.status,
            statusOutput: response.statusOutput,
            operations: response.operations,
            paymentOutput: response.paymentOutput,
            hostedCheckoutSpecificOutput: response.hostedCheckoutSpecificOutput,
        };
    },
};
